### PR TITLE
Add typings for indefinite

### PR DESCRIPTION
**Pull request by omaishar**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/indefinite/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Type definitions for indefinite 2.2
+// Project: https://github.com/tandrewnichols/indefinite
+// Definitions by: My Self <https://github.com/me>
```
Comment (RyanCavanaugh) [inline types/indefinite/index.d.ts]: Please replace the placeholder values with your name / username. Thanks!

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

**Review by amcasey — CHANGES_REQUESTED** [review]
Please add an owner.

File: types/indefinite/index.d.ts (hunk 1 above)
Comment (omaishar) [reply types/indefinite/index.d.ts]: Replaced

**Review by amcasey — APPROVED** [review]

**Comment by mankittens** [comment]
These types do not work for me! I'm getting this error from the typescript compiler:

```
error TS2349: Cannot invoke an expression whose type lacks a call signature. Type 'typeof import(".../app/node_modules/@types/indefinite/index")' has no compatible call signatures.
```

Here's the main JS file for indefinite.js
```js
const irregulars = require('./irregular-words');
const rules = require('./rules');

const indefinite = (word, opts = {}) => {
  let result;

  /**
   * I'd really prefer to use for of here, but babel converts that
   * to something using Symbol.iterator, which PhantomJS chokes on.
   */
  rules.some((rule) => {
    if (rule.check(word, opts)) {
      result = rule.run(word, opts);
      return true;
    }
  });

  return result;
};

indefinite.irregularWords = irregulars.list;

module.exports = indefinite;
```

And the types definition:
```js
export interface Options {
    capitalize?: boolean;
    caseInsensitive?: boolean;
    numbers?: "colloquial";
}
export function indefinite(word: string | number, opts?: Options): string;
```

Am I right in thinking that the types declaration is incorrect? The documentation for `indefinite` says to import like `import a from 'indefinite'`, but this produces an error in the compiler (when `@types/indefinite` is installed). The way to satisfy the compiler is to do `import { indefinite } from 'indefinite'`, but this produces the a runtime error like `indefinite.indefinite is undefined`.

**Pull request merged**